src/Entity/Components/CoinsComponent.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+namespace game
+{
+
+struct CoinsComponent
+{
+	int mMaxCoinAmount{100};
+	int mCoinAmount{};
```

Comment:
Usually for important variables I'll prefer to assign a value rather than relying on the default value. Not assigning one because you rely on the default value, can sometime give the impression to others that maybe something is missing